### PR TITLE
[Refactor:TAGrading] Enable collapsing in non-mobile view

### PR DESCRIPTION
**Pull request by DefinitiveAbove**
### What is the current behavior?
In the mobile view of the grading index for TA grading, collapsing grading sections is available. This is not available in the non-mobile view of the grading index.

### What is the new behavior?
In the non-mobile view of the grading index, collapsing/expanding grading sections is now available.

Expanded/collapsed grading section info is saved on a course-by-course basis in localStorage.

### CODE REVIEW

**Review by cjreed121 — CHANGES_REQUESTED** [review]
It functions well. I think it would be nice to have some sort of horizontal line or spacing between sections once they are minimized as it feels like all sections are just merged together once they are closed right now. Also it would be nice if when hovering over the arrow to toggle, your mouse cursor turned to a pointer so you could tell it does something.

**Review by cjreed121 — CHANGES_REQUESTED** [review]
Looks great but there is an issue with the auto collapse/expand. The cookie doesn't update for me so I never see it working. It seems cookies forbid special characters so I think the newline join is breaking it. Could probably just store a simple JSON object here. Also shouldn't this be local storage instead of a cookie? I think cookies should only be used if the server also needs to see and control its value.

**Comment by williamjallen** [comment]
> Also shouldn't this be local storage instead of a cookie? I think cookies should only be used if the server also needs to see and control its value.

Cookies allow the server to give the page an initial state (open or closed).  If local storage was used, the user would see a brief flash between the page initially rendering and the JS updating the relevant value from local storage and then re-rendering.

**Comment by cjreed121** [comment]
> > Also shouldn't this be local storage instead of a cookie? I think cookies should only be used if the server also needs to see and control its value.
> 
> Cookies allow the server to give the page an initial state (open or closed). If local storage was used, the user would see a brief flash between the page initially rendering and the JS updating the relevant value from local storage and then re-rendering.

Yeah I agree that could be beneficial in that case but we aren't using the cookies server side. So I think cookies should either be used server side or this should actually use local storage.

File: site/app/views/grading/ElectronicGraderView.php [hunk 1]
```diff
@@ -441,6 +441,8 @@ function sortTable(sort_element_index){
      * @return string
      */
     public function detailsPage(Gradeable $gradeable, $graded_gradeables, $teamless_users, $graders, $empty_teams, $show_all_sections_button, $show_import_teams_button, $show_export_teams_button, $show_edit_teams, $past_grade_start_date, $view_all, $sort, $direction, $anon_mode, $overrides, $anon_ids) {
+        $collapsed_sections = isset($_COOKIE['collapsed_sections']) ? explode("\n", $_COOKIE['collapsed_sections']) : [];
```
Comment (williamjallen) [inline site/app/views/grading/ElectronicGraderView.php:444]: Aren't the cookies supposed to be being used right here on the server side?  The value is then used in `Details.twig`.  I agree that the formatting will need to be changed to remove the newline for it to actually work.  Perhaps I misunderstood what you meant?

Comment (cjreed121) [reply site/app/views/grading/ElectronicGraderView.php]: Oh whoops, I thought this was all new code functionality. That all makes sense now for it to be cookies.

Comment (cjreed121) [reply site/app/views/grading/ElectronicGraderView.php]: Actually it is new functionality but I just skipped that line in my review before it seems.

**Review by cjreed121 — APPROVED** [review]
Works well!

**Pull request merged**